### PR TITLE
feat: set proxy_ssl_trusted_certificate as ssl_trusted_certificate

### LINKED ISSUE #6410: request help: mTLS Between APISIX and Upstream
### Issue description

1. I would like to ask mTLS Between APISIX and Upstream. If so, why is it that only the client certificate and key are configured when configuring the upstream, and the mTLS Between APISIX and Upstream can be completed without configuring the CA certificate？
2. 我通过抓包进行分析，发现客户端与apisix之间进行了一个完整且正常的双向认证，apisix与服务端之间也是进行一个完整的双向认证，但是apisix作为客户端与服务端进行双向认证是未校验服务端的证书，请问一下这样做的原因是什么？

### Environment

- apisix version (cmd: `apisix version`):v2.8
- OS (cmd: `uname -a`): linux centos7
- OpenResty / Nginx version (cmd: `nginx -V` or `openresty -V`):
- etcd version, if have (cmd: run `curl http://127.0.0.1:9090/v1/server_info` to get the info from server-info API):
- apisix-dashboard version, if have:
- the plugin runner version, if the issue is about a plugin runner (cmd: depended on the kind of runner):
- luarocks version, if the issue is about installation (cmd: `luarocks --version`):

### PR DESCRIPTION
**Pull request by tzssangglass**
Closes #6410

Signed-off-by: tzssangglass <tzssangglass@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

I want to add test cases to prove 

1. the upstream uses self-signed certificate, which is issued by the CA certificate set by `proxy_ssl_trusted_certificate`.
2. when APISIX proxy to upstream, the self-signed certificate is verified successfully after `proxy_ssl_trusted_certificate` is turned on, and no error like `18: self signed certificate` appears

I need some suggestions.

### CODE REVIEW

**Review by tokers — COMMENTED** [review]
We can add a test case to verify the SSL verification is really enabled or disabled. @tzssangglass

**Comment by tzssangglass** [comment]
> We can add a test case to verify the SSL verification is really enabled or disabled. @tzssangglass

added

File: t/cli/test_upstream_mtls.sh [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+fi
+
+sleep 0.1
+
+make stop
+
+echo "passed: connection to upstream with mtTLS failed success"
+
+# test proxy_ssl_trusted_certificate and use incorrect ca cert
+git checkout conf/config.yaml
```
Comment (spacewander) [inline t/cli/test_upstream_mtls.sh:92]: There is no need to check out it as we will write to it soon.

File: t/cli/test_upstream_mtls.sh [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+exit_if_not_customed_nginx
```
Comment (spacewander) [inline t/cli/test_upstream_mtls.sh:94]: No need to exit it twice

File: t/cli/test_upstream_mtls.sh [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        }
+    }
+}'
+
+sleep 0.1
+
+code=$(curl -v -k -i -m 20 -o /dev/null -s -w %{http_code} http://127.0.0.1:9080/hello)
+
+if [ ! $code -eq 200 ]; then
+    echo "failed: connection to upstream with mtTLS failed"
```
Comment (spacewander) [inline t/cli/test_upstream_mtls.sh:81]:
```suggestion
    echo "failed: connection to upstream with mTLS failed"
```

File: t/cli/test_upstream_mtls.sh (hunk 1 above)
Comment (tzssangglass) [reply t/cli/test_upstream_mtls.sh]: rm it

File: t/cli/test_upstream_mtls.sh (hunk 2 above)
Comment (tzssangglass) [reply t/cli/test_upstream_mtls.sh]: rm it

File: t/cli/test_upstream_mtls.sh (hunk 3 above)
Comment (tzssangglass) [reply t/cli/test_upstream_mtls.sh]: update

**Review by tokers — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**